### PR TITLE
[WIP] Remove createItems

### PR DESCRIPTION
**Pull request by timleslie**

### CODE REVIEW

**Comment by gautamsi** [comment]
I like the api `executeQuery` but I am sure keystone community would want a simple way to import initial data by creating array of object rather than wiring up all this. 

We may be able to wire up `createItems` in a way it uses `executeQuery` internally.

**Comment by timleslie** [comment]
> I like the api `executeQuery` but I am sure keystone community would want a simple way to import initial data by creating array of object rather than wiring up all this.
> 
> We may be able to wire up `createItems` in a way it uses `executeQuery` internally.

In the future we'll probably add convenience packages for doing bulk imports. This change is all about getting the APIs right for that. `createItems` has a similar but slightly different API for handling relationships when compared to the graphQL API, and this requires a significant amount of implementation to duplicate the work already done by the graphQL resolvers. This was useful during the initial development phase, as a way to easily populate the backend, but it is no longer relevant now that we have working resolvers.

As a general rule we want the core to support a single well defined way to get things done, and in this case that is the graphQL API.

File: test-projects/access-control/server.js [hunk 1]
```diff
@@ -16,14 +16,36 @@ keystone
       Object.values(keystoneApp.adapters).forEach(async adapter => {
         await adapter.dropDatabase();
       });
-      await keystoneApp.createItems(initialData);
+      await Promise.all(
+        Object.entries(initialData).map(async ([listName, items]) => {
+          const list = keystoneApp.lists[listName];
+          await keystoneApp.executeQuery({
```
Comment (jesstelford) [inline test-projects/access-control/server.js]:
```suggestion
          return keystoneApp.executeQuery({
```

File: test-projects/access-control/server.js [hunk 2]
```diff
@@ -16,14 +16,36 @@ keystone
       Object.values(keystoneApp.adapters).forEach(async adapter => {
         await adapter.dropDatabase();
       });
-      await keystoneApp.createItems(initialData);
+      await Promise.all(
+        Object.entries(initialData).map(async ([listName, items]) => {
```
Comment (jesstelford) [inline test-projects/access-control/server.js]:
```suggestion
        Object.entries(initialData).map(([listName, items]) => {
```

File: test-projects/access-control/server.js [hunk 3]
```diff
@@ -16,14 +16,36 @@ keystone
...
+      );
     }
 
     server.app.get('/reset-db', async (req, res) => {
       Object.values(keystoneApp.adapters).forEach(async adapter => {
         await adapter.dropDatabase();
       });
-      await keystoneApp.createItems(initialData);
+      await Promise.all(
+        Object.entries(initialData).map(async ([listName, items]) => {
```
Comment (jesstelford) [inline test-projects/access-control/server.js]:
```suggestion
        Object.entries(initialData).map(([listName, items]) => {
```

File: test-projects/access-control/server.js [hunk 4]
```diff
@@ -16,14 +16,36 @@ keystone
...
+          const list = keystoneApp.lists[listName];
+          await keystoneApp.executeQuery({
```
Comment (jesstelford) [inline test-projects/access-control/server.js]:
```suggestion
          return keystoneApp.executeQuery({
```

File: test-projects/basic/server.js [hunk 5]
```diff
@@ -16,14 +16,36 @@ keystone
       Object.values(keystoneApp.adapters).forEach(async adapter => {
         await adapter.dropDatabase();
       });
-      await keystoneApp.createItems(initialData);
+      await Promise.all(
+        Object.entries(initialData).map(async ([listName, items]) => {
```
Comment (jesstelford) [inline test-projects/basic/server.js]:
```suggestion
        Object.entries(initialData).map(([listName, items]) => {
```

File: test-projects/basic/server.js [hunk 6]
```diff
@@ -16,14 +16,36 @@ keystone
...
+          const list = keystoneApp.lists[listName];
+          await keystoneApp.executeQuery({
```
Comment (jesstelford) [inline test-projects/basic/server.js]:
```suggestion
          return keystoneApp.executeQuery({
```

File: test-projects/basic/server.js [hunk 7]
```diff
@@ -16,14 +16,36 @@ keystone
...
+      );
     }
 
     server.app.get('/reset-db', async (req, res) => {
       Object.values(keystoneApp.adapters).forEach(async adapter => {
         await adapter.dropDatabase();
       });
-      await keystoneApp.createItems(initialData);
+      await Promise.all(
+        Object.entries(initialData).map(async ([listName, items]) => {
```
Comment (jesstelford) [inline test-projects/basic/server.js]:
```suggestion
        Object.entries(initialData).map(([listName, items]) => {
```

File: test-projects/basic/server.js [hunk 8]
```diff
@@ -16,14 +16,36 @@ keystone
...
+          const list = keystoneApp.lists[listName];
+          await keystoneApp.executeQuery({
```
Comment (jesstelford) [inline test-projects/basic/server.js]:
```suggestion
          return keystoneApp.executeQuery({
```

File: test-projects/login/server.js [hunk 9]
```diff
@@ -18,14 +18,36 @@ keystone
       Object.values(keystoneApp.adapters).forEach(async adapter => {
         await adapter.dropDatabase();
       });
-      await keystoneApp.createItems(initialData);
+      await Promise.all(
+        Object.entries(initialData).map(async ([listName, items]) => {
```
Comment (jesstelford) [inline test-projects/login/server.js]:
```suggestion
        Object.entries(initialData).map(([listName, items]) => {
```

File: test-projects/login/server.js [hunk 10]
```diff
@@ -18,14 +18,36 @@ keystone
...
+          const list = keystoneApp.lists[listName];
+          await keystoneApp.executeQuery({
```
Comment (jesstelford) [inline test-projects/login/server.js]:
```suggestion
          return keystoneApp.executeQuery({
```

File: test-projects/login/server.js [hunk 11]
```diff
@@ -18,14 +18,36 @@ keystone
...
+      );
     }
 
     server.app.get('/reset-db', async (req, res) => {
       Object.values(keystoneApp.adapters).forEach(async adapter => {
         await adapter.dropDatabase();
       });
-      await keystoneApp.createItems(initialData);
+      await Promise.all(
+        Object.entries(initialData).map(async ([listName, items]) => {
```
Comment (jesstelford) [inline test-projects/login/server.js]:
```suggestion
        Object.entries(initialData).map(([listName, items]) => {
```

File: test-projects/login/server.js [hunk 12]
```diff
@@ -18,14 +18,36 @@ keystone
...
+          const list = keystoneApp.lists[listName];
+          await keystoneApp.executeQuery({
```
Comment (jesstelford) [inline test-projects/login/server.js]:
```suggestion
          return keystoneApp.executeQuery({
```

File: test-projects/social-login/server.js [hunk 13]
```diff
@@ -60,14 +60,36 @@ keystone
       Object.values(keystoneApp.adapters).forEach(async adapter => {
         await adapter.dropDatabase();
       });
-      await keystoneApp.createItems(initialData);
+      await Promise.all(
+        Object.entries(initialData).map(async ([listName, items]) => {
```
Comment (jesstelford) [inline test-projects/social-login/server.js]:
```suggestion
        Object.entries(initialData).map(([listName, items]) => {
```

File: test-projects/social-login/server.js [hunk 14]
```diff
@@ -60,14 +60,36 @@ keystone
...
+          const list = keystoneApp.lists[listName];
+          await keystoneApp.executeQuery({
```
Comment (jesstelford) [inline test-projects/social-login/server.js]:
```suggestion
          return keystoneApp.executeQuery({
```

File: test-projects/social-login/server.js [hunk 15]
```diff
@@ -60,14 +60,36 @@ keystone
...
+      );
     }
 
     server.app.get('/reset-db', async (req, res) => {
       Object.values(keystoneApp.adapters).forEach(async adapter => {
         await adapter.dropDatabase();
       });
-      await keystoneApp.createItems(initialData);
+      await Promise.all(
+        Object.entries(initialData).map(async ([listName, items]) => {
```
Comment (jesstelford) [inline test-projects/social-login/server.js]:
```suggestion
        Object.entries(initialData).map(([listName, items]) => {
```

File: test-projects/social-login/server.js [hunk 16]
```diff
@@ -60,14 +60,36 @@ keystone
...
+          const list = keystoneApp.lists[listName];
+          await keystoneApp.executeQuery({
```
Comment (jesstelford) [inline test-projects/social-login/server.js]:
```suggestion
          return keystoneApp.executeQuery({
```

**Comment by dcousens** [comment]
When merged, this probably closes https://github.com/keystonejs/keystone-5/issues/144

File: api-tests/auth-header.test.js [hunk 17]
```diff
@@ -106,7 +110,11 @@ multiAdapterRunners().map(({ runner, adapterName }) =>
         'Gives access denied when not logged in',
         runner(setupKeystone, async ({ keystone, server }) => {
           // seed the db
-          await keystone.createItems(initialData);
+          await keystone.executeQuery({
+            query: initialDataQuery,
+            schemaName: 'admin',
```
Comment (dcousens) [inline api-tests/auth-header.test.js]: What does `schemaName` refer to?

**Comment by jesstelford** [comment]
This _may_ be replaceable by the Programatic access, specifically an example like: https://github.com/keystonejs/keystone-5/issues/1275#issuecomment-510274346

**Comment by timleslie** [comment]
One day this feature will land, but when it does, it won't be on this PR 🤷‍♂

**Pull request closed without merging**